packages/@react-spectrum/s2/src/ActionButton.tsx

```diff
@@ -281,6 +311,10 @@ export const ActionButton = forwardRef(function ActionButton(props: ActionButton
           [AvatarContext, {
             size: avatarSize[size],
             styles: style({marginStart: '--iconMargin', flexShrink: 0, order: 0})
+          }],
+          [NotificationBadgeContext, {
+            size: props.size === 'XS' ? undefined : props.size,
+            styles: style({position: 'absolute', top: '--badgeTop', insetStart: '[var(--badgePosition)]', marginTop: '[calc((self(height) * -1)/2)]', marginStart: '[calc((self(height) * -1)/2)]'})
```

Comment:
I wonder if there should be disabled styles as well if the ActionButton is disabled but has a notification. Maybe the notification shouldn't even appear? Also if the ActionButton is quiet and only has text, the badge feels a bit far from the text IMO (though maybe it is obvious that it is still associated in real world cases), might be worth bringing up with design

Reply:
good catch on quiet, that should definitely be brought up with design

Reply:
Might be a little simpler to do `translate: '[-50%']` here instead of using margins. Percentages in translate are relative to the size.

Reply:
hm we could maybe. i think the issue is mainly:
1) translate doesn't exist on styles. we'd have to use UNSAFE_style which i'd like to avoid 

2) probably the bigger issue of the two. the way that the indicator is positioned is reliant on its height because its constant. the width of the badge is variable. the way i think of it, each badge has an invisible circle along the rounded corner, and we want the corner of the svg to align with the center of this circle. if we were to use translateX(-50%), it'd position itself such that the corner of the svg is aligned with the middle of the whatever the width of the badge is. maybe that'd help with the visual collisions though so i guess we can see what design says about that